pkg/cmdimpl/add_test.go

```diff
@@ -502,4 +500,58 @@ var _ = Describe("Wet Run Add Test", func() {
 				status.Override(shandler))
 		})
 	})
+
+	It("fails when the repository is in a dirty state", func() {
```

Comment:
To me this feels a bit like http://xunitpatterns.com/Obscure%20Test.html

It's really not clear what it's doing?

Could we maybe extract some functions to clarify what's going on?

Reply:
this is not relevant anymore, I'll move this PR out to the new packages that were created after the refactoring.